domain/entities/transfers/transfers.go

```diff
@@ -0,0 +1,36 @@
+package transfers
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	errInvalidLengthID = errors.New("Empty ID not allowed")
+	errInvalidAmount   = errors.New("Invalid amount, be must > 0")
+	errEqualAccounts   = errors.New("Equal source and destination accounts are not allowed.")
+)
+
+type Transfer struct {
+	ID                     string    `json:"id"`
+	Account_origin_ID      string    `json:"account_origin_id"`
+	Account_destination_ID string    `json:"account_destination_id"`
+	Amount                 uint      `json:"amount"`
+	Created_At             time.Time `json:"created_at"`
+}
+
+func ValidateTransferData(transfer *Transfer) error {
```

Comment:
```suggestion
func ValidateTransferData(transfer Transfer) error {
```
Como você não está alterando a `transfer`, não necessita de ponteiro. O que acha?

Salvaguarda importante: Com structs MUITO grandes é comum passarmos por referência, como você fez aí, por conta do tamanho da cópia, já que é mais rápido copiar os 64bits do endereço do que todos os fields da struct. Porém nesse caso parece desnecessário e pode passar a ideia de que a transfer está sofrendo alterações.